hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1689,4 +1696,10 @@ private InternalSchema getInternalSchema(TableSchemaResolver schemaUtil) {
       }
     });
   }
+
+  private void validateComplexKeygen(HoodieTableConfig tableConfig) {
+    if (isComplexKeyGeneratorWithSingleRecordKeyField(tableConfig)) {
```

Comment:
do you think we also need to check the write config flag? if user sets it as false explicitly.